app/server/dao/src/main/java/io/syndesis/server/dao/validation/TargetWithVector.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.server.dao.validation;
+
+import java.util.Collection;
+import java.util.Optional;
+import io.syndesis.common.model.Kind;
+import io.syndesis.common.model.WithId;
+
+public class TargetWithVector<T extends WithId<T>> implements WithId<T> {
+
+    private T target;
+
+    private Collection<T> vector;
+
+    public TargetWithVector(T target, Collection<T> vector) {
+        this.target = target;
+        this.vector = vector;
+    }
+
+    public T getTarget() {
+        return target;
+    }
+
+    public Collection<T> getVector() {
+        return vector;
+    }
```

Comment:
Not a huge fan of "Vector", maybe we can use `Stream<T> stream()~

Reply:
This is not the java Vector object but the technical term vector as in "I have a seach vector". In that sense stream doesn't make much sense. Also, Stream can only be used once then it is closed, which I think might be a little limiting?

Reply:
The usage pattern seems always like "foreach" thus I was thinking about using stream, but that's not a huge issue.